### ISSUE TITLE
certain forms of "require" like dependency not detected by Clerk

### ISSUE THREAD

**Issue by behrica** [nextjournal/clerk#117]
In using `require-python`  from `libpython-clj` teh dependency detection is not working: 

```clojure
(require-python '[os :as os])
(os/getcwd)
```

fails with "os not found"

**Comment by behrica** [comment]
Same for "import":

```
(import java.awt.image.BufferedImage
        java.awt.Color
        sun.java2d.SunGraphics2D)
        
```

**Comment by mk** [comment]
Yep, the `require-python` uses `intern` eventually which makes it hard to detect via static analysis. Thinking we could do something based on `with-redefs` however or at least detect that something depends on `intern` and warn.

The `import` issue is different and should be easily solvable. Feel free to move that to a separate issue.

**Comment by mk** [comment]
I tried reproducing this but ran into the error below.

I also believe this has been fixed with #166. @behrica Please give it a try and reopen if it's still an issue, ideally with a repo in which I we can reproduce it.
```
#error {
 :cause "namespace 'libpython-clj2.metadata' not found"
 :via
 [{:type clojure.lang.Compiler$CompilerException
   :message "Syntax error compiling at (libpython_clj2/require.clj:1:1)."
   :data #:clojure.error{:phase :compile-syntax-check, :line 1, :column 1, :source "libpython_clj2/require.clj"}
   :at [clojure.core$throw_if invokeStatic "core.clj" 5892]}
  {:type java.lang.Exception
   :message "namespace 'libpython-clj2.metadata' not found"
   :at [clojure.core$apply invokeStatic "core.clj" 669]}]
 :trace
 [[clojure.core$apply invokeStatic "core.clj" 669]
  [clojure.core$load_libs invokeStatic "core.clj" 6016]
  [clojure.core$load_libs doInvoke "core.clj" 6000]
  [clojure.lang.RestFn applyTo "RestFn.java" 137]
  [clojure.core$apply invokeStatic "core.clj" 669]
  [clojure.core$require invokeStatic "core.clj" 6038]
  [clojure.core$require doInvoke "core.clj" 6038]
  [clojure.lang.RestFn invoke "RestFn.java" 482]
  [libpython_clj2.require$eval69797$loading__6789__auto____69798 invoke "require.clj" 1]
  [libpython_clj2.require$eval69797 invokeStatic "require.clj" 1]
  [libpython_clj2.require$eval69797 invoke "require.clj" 1]
  [clojure.lang.Compiler eval "Compiler.java" 7194]
  [clojure.lang.Compiler eval "Compiler.java" 7183]
  [clojure.lang.Compiler load "Compiler.java" 7653]
  [clojure.lang.RT loadResourceScript "RT.java" 381]
  [clojure.lang.RT loadResourceScript "RT.java" 372]
  [clojure.lang.RT load "RT.java" 459]
  [clojure.lang.RT load "RT.java" 424]
  [clojure.core$load$fn__6908 invoke "core.clj" 6161]
  [clojure.core$load invokeStatic "core.clj" 6160]
  [clojure.core$load doInvoke "core.clj" 6144]
  [clojure.lang.RestFn invoke "RestFn.java" 408]
  [clojure.core$load_one invokeStatic "core.clj" 5933]
  [clojure.core$load_one invoke "core.clj" 5928]
  [clojure.core$load_lib$fn__6850 invoke "core.clj" 5975]
  [clojure.core$load_lib invokeStatic "core.clj" 5974]
  [clojure.core$load_lib doInvoke "core.clj" 5953]
  [clojure.lang.RestFn applyTo "RestFn.java" 142]
  [clojure.core$apply invokeStatic "core.clj" 669]
  [clojure.core$load_libs invokeStatic "core.clj" 6016]
  [clojure.core$load_libs doInvoke "core.clj" 6000]
  [clojure.lang.RestFn applyTo "RestFn.java" 137]
  [clojure.core$apply invokeStatic "core.clj" 669]
  [clojure.core$require invokeStatic "core.clj" 6038]
  [clojure.core$require doInvoke "core.clj" 6038]
  [clojure.lang.RestFn invoke "RestFn.java" 408]
  [scratch$eval69789$loading__6789__auto____69790 invoke "NO_SOURCE_FILE" 1]
  [scratch$eval69789 invokeStatic "NO_SOURCE_FILE" 1]
  [scratch$eval69789 invoke "NO_SOURCE_FILE" 1]
  [clojure.lang.Compiler eval "Compiler.java" 7194]
  [clojure.lang.Compiler eval "Compiler.java" 7183]
  [clojure.lang.Compiler eval "Compiler.java" 7149]
  [clojure.core$eval invokeStatic "core.clj" 3215]
  [clojure.core$eval invoke "core.clj" 3211]
  [nrepl.middleware.interruptible_eval$evaluate$fn__17986$fn__17987 invoke "interruptible_eval.clj" 87]
  [clojure.lang.AFn applyToHelper "AFn.java" 152]
  [clojure.lang.AFn applyTo "AFn.java" 144]
  [clojure.core$apply invokeStatic "core.clj" 667]
  [clojure.core$with_bindings_STAR_ invokeStatic "core.clj" 1990]
  [clojure.core$with_bindings_STAR_ doInvoke "core.clj" 1990]
  [clojure.lang.RestFn invoke "RestFn.java" 425]
  [nrepl.middleware.interruptible_eval$evaluate$fn__17986 invoke "interruptible_eval.clj" 87]
  [clojure.main$repl$read_eval_print__9206$fn__9209 invoke "main.clj" 437]
  [clojure.main$repl$read_eval_print__9206 invoke "main.clj" 437]
  [clojure.main$repl$fn__9215 invoke "main.clj" 458]
  [clojure.main$repl invokeStatic "main.clj" 458]
  [clojure.main$repl doInvoke "main.clj" 368]
  [clojure.lang.RestFn invoke "RestFn.java" 1523]
  [nrepl.middleware.interruptible_eval$evaluate invokeStatic "interruptible_eval.clj" 84]
  [nrepl.middleware.interruptible_eval$evaluate invoke "interruptible_eval.clj" 56]
  [nrepl.middleware.interruptible_eval$interruptible_eval$fn__18019$fn__18023 invoke "interruptible_eval.clj" 152]
  [clojure.lang.AFn run "AFn.java" 22]
  [nrepl.middleware.session$session_exec$main_loop__18089$fn__18093 invoke "session.clj" 218]
  [nrepl.middleware.session$session_exec$main_loop__18089 invoke "session.clj" 217]
  [clojure.lang.AFn run "AFn.java" 22]
  [java.lang.Thread run "Thread.java" 833]]}
```
